core/urls.py

```diff
@@ -13,18 +13,18 @@
     1. Add a URL to urlpatterns:  url(r'^blog/', include('blog.urls'))
 """
 from django.conf.urls import url, include
-from django.contrib import admin
 from django.conf import settings
 from django.conf.urls.static import static
 
 from geonode_generic.urls import urlpatterns as geonode_generic_urlpatterns
 
+# GeoNode has to be in root url conf.
+# It cannot be included in include() function because it contains i18n urls
```

Comment:
Can't do anything about this for now.
GeoNode will be in the root url.
BIMS have to be in sub url.

Reply:
Yes we can have a custom nginx rule to reroute /geonode suburl to bims root url (because we can't change the root url of geonode+bims directly.
My concern is that the rerouting won't be visible from inside django. All internal bims django resover will resolve to root url.